### PR TITLE
dts: st: pinctrl.dtsi files including DCMI pins

### PR DESCRIPTION
**Pull request by CharlesDias**
Generate DCMI pins definitions for H7 family.

Related PR:

- https://github.com/zephyrproject-rtos/zephyr/pull/71462
- https://github.com/zephyrproject-rtos/zephyr/pull/71463

### CODE REVIEW

File: scripts/genpinctrl/stm32-pinctrl-config.yaml [hunk 1]
```diff
@@ -52,6 +52,50 @@
...
+  match: "^DCMI_D5$"
+  slew-rate: very-high-speed
+
+- name: DCMI_D6
+  match: "^DCMI_D6$"
+  slew-rate: very-high-speed
+
+- name: DCMI_D7
+  match: "^DCMI_D7$"
+  slew-rate: very-high-speed
```
Comment (erwango) [inline scripts/genpinctrl/stm32-pinctrl-config.yaml:97]: Could you use a regular expression ?
See for instance:
```
- name: FMC
  match: "^FMC_(?:NL|NADV|CLK|NBL[0-3]|A\\d+|D\\d+|NE[1-4]|NOE|NWE|NWAIT|NCE|INT|SDCLK|SDNWE|SDCKE[0-1]|SDNE[0-1]|SDNRAS|SDNCAS)$"
```
Aside from conciseness, this is slightly faster to generate.

**Review by erwango — CHANGES_REQUESTED** [review]

File: scripts/genpinctrl/stm32-pinctrl-config.yaml (hunk 1 above)
Comment (CharlesDias) [reply scripts/genpinctrl/stm32-pinctrl-config.yaml]: Done

**Comment by erwango** [comment]
Looks good to me but would it be possible to generate pins as well for other series?
I know driver is H7 specific for now (at least only validated on H7), but support might be extended in future. Having pins already available would be a good start.
This is the way we're proceeding in general.

**Comment by CharlesDias** [comment]
Hi, @erwango.

I have two questions about the files generated after run the Python script:
1. Do I have to add also the new pinctrl files related to STM32H523, STM32H533, STM32H7R, and STM32H7s families? Those files are totally new.
2. I noticed that in `stm32h725iekx-pinctrl.dtsi`, `stm32h725igkx-pinctrl.dtsi`, `stm32h730ibkxq-pinctrl.dtsi`, and `stm32h735igkx-pinctrl.dtsi` files, the `adc3_inp16_ph5` was removed. Do I have to accept this modification? I'll check if something similar happens with other files.

**Comment by erwango** [comment]
Thanks for raising this @CharlesDias.
This probably comes from the fact that you're generating from a latest version of open_pin_data repo.
2 options:
- Keep working with this commit, but then do it in 2 passes: first new batch from the new version w/o dcmi files, then add the new pinctrl, generate once again and provide the new signals in a 3 commit. In this case, don't forget to update commit used in https://github.com/zephyrproject-rtos/hal_stm32/blob/main/dts/README.rst (or maybe it is done automatically)
- Generate from SHA1 version provided in https://github.com/zephyrproject-rtos/hal_stm32/blob/main/dts/README.rst. And let the  open_pin_data update aside for now.

**Comment by CharlesDias** [comment]
Hi, @erwango.

I choose the second option. ;-)

**Comment by erwango** [comment]
This is fine for me now, but please update the first commit title to remove H7 reference, as pr is stm32 generic now.

**Review by erwango — APPROVED** [review]

**Pull request merged**